sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/SqlOperators.java

```diff
@@ -92,6 +93,14 @@
           x -> createTypeFactory().createArrayType(x.getOperandType(0), -1),
           new UdafImpl<>(new ArrayAgg.ArrayAggArray()));
 
+  public static final SqlOperator ARRAY_CONCAT_AGG_FN =
+      createUdafOperator(
+          "array_concat_agg",
+          x ->
+              createTypeFactory()
+                  .createArrayType(createTypeFactory().createSqlType(SqlTypeName.BIGINT), -1),
```

Comment:
You need to get the array element type from `x`. Otherwise this function will only work for integers.

Reply:
That was really a huge bug you identified. Thanks @ibzib :)